clients/clients-integration-tests/src/test/java/org/apache/kafka/clients/consumer/PlaintextConsumerCommitTest.java

```diff
@@ -0,0 +1,680 @@
+        testAutoCommitOnCloseAfterWakeup(GroupProtocol.CONSUMER);
+    }
+
+    private void testAutoCommitOnCloseAfterWakeup(GroupProtocol groupProtocol) throws InterruptedException {
+        try (var consumer = createConsumer(groupProtocol, true)) {
+            sendRecords(10000);
+
+            consumer.subscribe(List.of(topic));
+            awaitAssignment(consumer, Set.of(tp, tp1));
+
+            // should auto-commit sought positions before closing
+            consumer.seek(tp, 300);
+            consumer.seek(tp1, 500);
+
+            // wakeup the consumer before closing to simulate trying to break a poll
+            // loop from another thread
+            consumer.wakeup();
+        }
+
+        // now we should see the committed positions from another consumer
+        try (var anotherConsumer = createConsumer(groupProtocol, true)) {
+            assertEquals(300, anotherConsumer.committed(Set.of(tp)).get(tp).offset());
+            assertEquals(500, anotherConsumer.committed(Set.of(tp1)).get(tp1).offset());
+        }
+    }
+
+    @ClusterTest
+    public void testClassicConsumerCommitMetadata() throws InterruptedException {
+        testCommitMetadata(GroupProtocol.CLASSIC);
+    }
+
+    @ClusterTest
+    public void testAsyncConsumerCommitMetadata() throws InterruptedException {
+        testCommitMetadata(GroupProtocol.CONSUMER);
+    }
+
+    private void testCommitMetadata(GroupProtocol groupProtocol) throws InterruptedException {
+        try (var consumer = createConsumer(groupProtocol, true)) {
+            consumer.assign(List.of(tp));
+            // sync commit
+            var syncMetadata = new OffsetAndMetadata(5, Optional.of(15), "foo");
+            consumer.commitSync(Map.of(tp, syncMetadata));
+            assertEquals(syncMetadata, consumer.committed(Set.of(tp)).get(tp));
+
+            // async commit
+            var asyncMetadata = new OffsetAndMetadata(10, "bar");
+            sendAndAwaitAsyncCommit(consumer, Optional.of(Map.of(tp, asyncMetadata)));
+            assertEquals(asyncMetadata, consumer.committed(Set.of(tp)).get(tp));
+
+            // handle null metadata
+            var nullMetadata = new OffsetAndMetadata(5, null);
+            consumer.commitSync(Map.of(tp, nullMetadata));
+            assertEquals(nullMetadata, consumer.committed(Set.of(tp)).get(tp));
+        }
+    }
+
+    @ClusterTest
+    public void testClassicConsumerAsyncCommit() throws InterruptedException {
+        testAsyncCommit(GroupProtocol.CLASSIC);
+    }
+
+    @ClusterTest
+    public void testAsyncConsumerAsyncCommit() throws InterruptedException {
+        testAsyncCommit(GroupProtocol.CONSUMER);
+    }
+
+    private void testAsyncCommit(GroupProtocol groupProtocol) throws InterruptedException {
+        cluster.createTopic(
+            Topic.GROUP_METADATA_TOPIC_NAME, 
+            Integer.parseInt(OFFSETS_TOPIC_PARTITIONS), 
+            Short.parseShort(OFFSETS_TOPIC_REPLICATION)
+        );
```

Comment:
Could you add some comments about why we need to manually create internal topic here? It looks like we don't set `AUTO_CREATE_TOPICS_ENABLE_CONFIG` as `false`.